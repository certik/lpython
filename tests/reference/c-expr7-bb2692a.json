{
    "basename": "c-expr7-bb2692a",
    "cmd": "lpython --no-color --show-c {infile}",
    "infile": "tests/expr7.py",
    "infile_hash": "4a455e2279eb7dd269d81c1a257dea625b17b100d92e304e5ac34421",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "c-expr7-bb2692a.stdout",
    "stdout_hash": "1300b234a8ba0b61ca9b134d106f05f2d23538aa3b7bcd4c308e6e47",
    "stderr": "c-expr7-bb2692a.stderr",
    "stderr_hash": "28509dd59a386eebd632340a550d14299cd2a921ef6dc3ac7dbe7fe9",
    "returncode": 0
}