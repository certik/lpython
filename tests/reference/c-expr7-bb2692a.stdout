#include <assert.h>
#include <complex.h>
#include <inttypes.h>
#include <stdio.h>
#include <math.h>

#include <lfortran_intrinsics.h>

void _lpython_main_program()
{
    main0();
}

void main0()
{
    int32_t c;
    test_pow();
    c = test_pow_1(1, 2);
}

void test_pow()
{
    int32_t a;
    a = __lpython_overloaded_0__pow(2, 2);
}

int test_pow_1(int32_t a, int32_t b)
{
    int32_t _lpython_return_variable;
    int32_t res;
    res = __lpython_overloaded_0__pow(a, b);
    _lpython_return_variable = res;
    return _lpython_return_variable;
}

int __lpython_overloaded_0__pow(int32_t x, int32_t y)
{
    int32_t _lpython_return_variable;
    _lpython_return_variable = pow(x, y);
    return _lpython_return_variable;
}

float _lfortran_caimag(float complex x);

double _lfortran_zaimag(double complex x);

int main(int argc, char* argv[])
{
        _lpython_main_program();
    return 0;
}
